src/authService.js

```diff
@@ -63,28 +83,97 @@ export class AuthService {
 
   login(email, password) {
     let loginUrl = this.auth.getLoginUrl();
-    let content;
+    let config = this.config;
+    let clientId = this.config.clientId;
+    let content = {};
+    let options = {};
+    let data = [];
     if (typeof arguments[1] !== 'string') {
       content = arguments[0];
     } else {
-      content = {
+      content = clientId ?  {
+        'email': email,
+        'password': password,
+        'client_id': clientId
+      }
+      : {
         'email': email,
         'password': password
       };
     }
 
-    return this.client.post(loginUrl, content)
+    if (this.config.postContentType === 'json') {
+      content = JSON.stringify(content);
+    } else if (this.config.postContentType === 'form') {
+      for (let key in content) {
+        data.push(key + '=' + content[key]);
+      }
+      content = data.join('&');
+      options = {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded'
+        }
+      };
+    }
+
+    return this.client.post(loginUrl, content, options)
       .then(response => {
         this.auth.setTokenFromResponse(response);
-
+        if (config.useRefreshToken) {
+          this.auth.setRefreshTokenFromResponse(response);
+        }
         return response;
       });
   }
-
   logout(redirectUri) {
     return this.auth.logout(redirectUri);
   }
 
+  updateToken() {
+    this.isRefreshing = true;
+    let loginUrl = this.auth.getLoginUrl();
+    let refreshToken = this.auth.getRefreshToken();
+    let clientId = this.config.clientId;
+    let content = {};
+    let data = [];
+    let options = {};
+    if (refreshToken) {
+      content = clientId ? {
```

Comment:
This is the second time I see this construction, which makes me think, perhaps it's nicer to do something like this:

```js
var content = {foo: '', bar: ''};

if (clientId) {
  content['client_id'] = clientId;
}
```

This is not a complaint, more of a preference.